# BBS-230B (2023)

## Bulk RNA-seq, June 14, 2023

### Learning objectives

* Explain common considerations when designing a bulk RNA-seq experiment
* List the steps involved in the analysis of a bulk RNA-seq dataset
* Discuss common challenges and how to overcome them

### Lessons
1. [Experimental design](https://hbctraining.github.io/DGE_workshop_salmon_online/lessons/experimental_planning_considerations.html) (20 min)
1. [Raw data to counts workflow](https://hbctraining.github.io/DGE_workshop_salmon_online/lessons/01a_RNAseq_processing_workflow.html) (25 min)
2. [Differential gene expression analysis](https://hbctraining.github.io/rnaseq-cb321/lessons/dge_analysis.html) (25 min)
5. [Functional analysis](lessons/functional_analysis.md) (20 min)
6. [Visualization of DE results](https://hbctraining.github.io/Training-modules/planning_successful_rnaseq/lessons/data_visualization.html) (time permitting)


* [Understanding PCA](https://hbctraining.github.io/scRNA-seq_online/lessons/05_theory_of_PCA.html)

## Single cell RNA-seq, June 15, 2023

### Learning objectives

* Explain common considerations when designing a single-cell RNA-seq experiment
* List the steps involved in the analysis of a single cell dataset
* List the key statistical concepts utilized for the analysis

### Lessons
* [Single cell RNA-seq](../slides/Intro-to-scRNAseq.pdf) (50 mins)

## Resources (bulk RNA-seq)

* [bulk RNA-seq "Part I" (FASTQ to count matrix) workshop](https://hbctraining.github.io/Intro-to-rnaseq-hpc-salmon-flipped/)
* [Differential Gene Expression analysis workshop](https://hbctraining.github.io/DGE_workshop_salmon_online/)
* [Planning a successfull bulk RNA-seq experiment](https://hbctraining.github.io/Training-modules/planning_successful_rnaseq/#contents)
* [DESeq2 vignette](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#theory-behind-deseq2)
* [Functional analysis visualization](https://yulab-smu.top/biomedical-knowledge-mining-book/enrichplot.html)
* [Ggplot2 for functional analysis](https://hbctraining.github.io/Training-modules/Tidyverse_ggplot2/lessons/ggplot2.html)

## Resources (Single cell RNA-seq)

* [Single-cell RNA-seq workshop](https://hbctraining.github.io/scRNA-seq/)
* ["How many cells are needed per sample for my single-cell experiment?"](https://satijalab.org/howmanycells/)
* [http://bioconductor.org/books/release/OSCA/](http://bioconductor.org/books/release/OSCA/)
* [https://liulab-dfci.github.io/bioinfo-combio/](https://liulab-dfci.github.io/bioinfo-combio/)
* [https://hemberg-lab.github.io/scRNA.seq.course/](https://hemberg-lab.github.io/scRNA.seq.course/)
* [https://github.com/SingleCellTranscriptomics](https://github.com/SingleCellTranscriptomics)
* [Seurat vignettes](https://satijalab.org/seurat/vignettes.html)

## Resources (general)
* [RMarkdown](https://hbctraining.github.io/Training-modules/Rmarkdown/)
* ["Principal Component Analysis (PCA) clearly explained"](https://www.youtube.com/watch?v=_UVHneBUBW0), a video from [Josh Starmer](https://twitter.com/joshuastarmer)
